client.go

```diff
@@ -489,17 +489,32 @@ func (c *Client) websocket(operation string, secret string) (*websocket.Conn, er
 }
 
 func (c *Client) url(elem ...string) string {
+	// Normalize the URL
 	path := strings.Join(elem, "/")
+	entries := []string{}
```

Comment:
Bummer.  The only other thing I can think of would be to check for `//` in a loop.  The optimized version would look slightly weird but I think the intent would still be easier to understand than Join+Split+Join.  Most scenarios would be faster, only cases with lots of redundant slashes would be slower.

```
path := strings.Join(elem, "/")
for path = strings.Replace(path, "//", "/", -1); strings.Contains(path, "//"); path = strings.Replace(path, "//", "/", -1) {
}
```

Less optimized but easier to read version:

```
path := strings.Join(elem, "/")
for strings.Contains(path, "//") {
	path = strings.Replace(path, "//", "/", -1)
}
```